### ISSUE TITLE
identification.disagreement not getting updated when taxonomy changes

### ISSUE THREAD

**Issue by loarie** [inaturalist/inaturalist#1613]
This is happening if the taxonomy changes, e.g. if the identification was made when the disagree taxon was sibling to the observation.taxon and the disagree taxon was later grafted as a descendant of the observation taxon as in https://www.inaturalist.org/observations/9444854
<img width="535" alt="screen shot 2018-01-13 at 5 19 46 pm" src="https://user-images.githubusercontent.com/598026/34911778-fd4dc988-f885-11e7-86c6-3a19451d2ff5.png">

Proposed solution:
* when a taxon's ancestry changes, revisit all identifications of it and its descendants that are disagreements
* if a disagreement's `previous_observation_taxon` is an ancestor of the identification's taxon
  * set `disagreement` to `false`
  * update the Observation Taxon and Community Taxon for the observation

**Comment by loarie** [comment]
Another example from the google group https://www.inaturalist.org/observations/8191088

**Comment by loarie** [comment]
Here's another one relayed to the GoogleGroup
https://www.inaturalist.org/observations/3093248
Where user A: species A, user B: Genus C (disagree with Species A) got corrupted as:
<img width="947" alt="screen shot 2018-01-15 at 9 58 37 am" src="https://user-images.githubusercontent.com/598026/34955757-b80ea620-f9da-11e7-8df1-5f30b03143fe.png">
Removing the ID and re-adding it gets it right
<img width="964" alt="screen shot 2018-01-15 at 9 57 55 am" src="https://user-images.githubusercontent.com/598026/34955768-bf5cd8ac-f9da-11e7-9e47-183d4e1d7ffc.png">
Any idea what glitch is causing this?

**Comment by tonyrebelo** [comment]
Here is another example.  Note how the classification of the disagreement changes, but note most especially my ID is not a disagreement at all AS STATED!!
https://www.inaturalist.org/observations/9432621

Note:
3 people posted exactly the same thing in sequence: 
* rjq disagrees this is Cape White-eye (nominate ssp) Zosterops capensis ssp. capensis
* tonyrebelo disagrees this is Cape White-eye Zosterops capensis
* colin does not disagree.

BUT most importantly, the ID posted is a subspecies of Zosterops capensis: so it is not a disagreement to its own species!

**Comment by zookanthos** [comment]
1. Explicitly disagreed with ID. Everything works as expected (at least that I noticed).
2. Edited ID to fix a typo.
3. Now it is saying I disagreed with myself. The community ID is correct, but the label still says Leucobryum glaucum and the observer is not rejecting community IDs.
4. Withdraw observation. Now it says I disagree it is Leucobryum glaucum in the withdrawn ID.
5. Resubmit my ID, and everything works as expected.

https://www.inaturalist.org/observations/8810308

![capture](https://user-images.githubusercontent.com/35535478/35053450-59a713ae-fb67-11e7-9312-79a23ba2ebb3.PNG)
